### PR TITLE
added support for parsing esp packets

### PR DESCRIPTION
**Pull request by danwgun**

### CODE REVIEW

File: PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            get { return ReadUInt(Offset.SequenceNumber, Endianity.Big); }
+        }
+
+        /// <summary>
+        /// The payload of the ESP datagram.
+        /// </summary>
+        public Datagram Payload
+        {
+            get { return new Datagram(Buffer, StartOffset + HeaderLength, Length - HeaderLength); }
```
Comment (bricknerb) [inline PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs:79]: Wouldn't that include the pading, pad length, next header and ICV?

**Review by bricknerb — CHANGES_REQUESTED** [review]
1. Could you add tests? See GreTests and WiresharkCompareTests.
2. Could you add EspLayer?

I have different nits to make the style conform better with the rest of the code, let me know if you'd like me to add comments or I can just fix it after this is merged.

File: PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs (hunk 1 above)
Comment (danwgun) [reply PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs:79]: > 1. Could you add tests? See GreTests and WiresharkCompareTests.
> 2. Could you add EspLayer?
> 
> I have different nits to make the style conform better with the rest of the code, let me know if you'd like me to add comments or I can just fix it after this is merged.

Thanks for checking out my PR. I'll continue working on this.

Comment (danwgun) [reply PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs:79]: > Wouldn't that include the pading, pad length, next header and ICV?

I tried to replicate the way Wireshark parses these packets. It includes all of this data in the payload.

Comment (bricknerb) [reply PcapDotNet/src/PcapDotNet.Packets/Esp/EspDatagram.cs:79]: Great, let me know when you want me to take another look.

**Pull request closed without merging**